libtiledbvcf/src/stats/variant_stats.cc

```diff
@@ -463,16 +492,39 @@ void VariantStats::process(
     }
   }
 
+  bool is_nr_block;
+  {
+    bool one_alt = rec->n_allele == 2;
+    // if no first alt, or if wrong number of alts, this will be blank:
+    // no need to check whether this be a ref block, because this bool will be
+    // used inside an if statement
+    auto alt = one_alt ? alt_string(ref, rec->d.allele[1]) : "";
+    is_nr_block = (alt == "<NON_REF>");
+  }
+
   bool already_added_homozygous = false;
   for (int i = 0; i < ngt; i++) {
+    int length = end_pos - pos + 1;
+    if ((length = end_pos - pos + 1) > max_length_)
+      max_length_ = length;
     // If not missing, update allele count for GT[i]
     if (!gt_missing[i]) {
       auto alt = alt_string(ref, rec->d.allele[gt[i]]);
 
       if (gt[i] == 0) {
-        values_["ref"][AC] += count_delta_;
+        std::string ref_key = "ref";
+        if (is_nr_block) {
+          // ref block
+          ref_key = "nr";
+          values_["nr"][CUM_MAX] = max_length_;
```

Comment:
Is this case, max_length will be the max length of ref blocks. Should we keep track of the longest record, including variants and ref blocks?